src/main/java/Task.java

```diff
@@ -0,0 +1,41 @@
+public abstract class Task {
+    private String task;
+    private boolean isCompleted;
+
+    public Task(String task) throws IllegalArgumentException {
+        if (task.isEmpty() || task.equals(" ")){
+            throw new IllegalArgumentException("\t Oh no!!(0o0) Description cannot be empty");
+        }
+        this.task = task;
+        this.isCompleted = false;
+    }
+
+    public Task(String task, Boolean isCompleted) throws IllegalArgumentException {
+        if (task.isEmpty() || task.equals(" ")){
+            throw new IllegalArgumentException("\t Oh no!!(0o0) Description cannot be empty");
+        }
+        this.task = task;
+        this.isCompleted = isCompleted;
+    }
+
+    public String setCompleted(boolean completed) {
+        if (this.isCompleted == completed) {
+            throw new IllegalArgumentException("\t It seems the task has already been marked as such");
+        }
```

Comment:
Perhaps boolean variables/methods should be named to sound like booleans, e.g. isCompleted